types/src/v0/impls/state.rs

```diff
@@ -316,15 +316,16 @@ pub fn validate_proposal(
         });
     }
 
-    // Check if timestamp is increasing (with some tolerance).
-    if proposal.timestamp() <= parent_header.timestamp() - 2 {
+    // Check if timestamp is increasing.
+    let system_time: u64 = OffsetDateTime::now_utc().unix_timestamp() as u64;
+    if proposal.timestamp() < system_time - 2 {
         return Err(ProposalValidationError::InvalidTimestamp {
-            parent_timestamp: parent_header.timestamp(),
             proposal_timestamp: proposal.timestamp(),
+            local_timestamp: proposal.timestamp(),
```

Comment:
Should this be `system_time`?

Reply:
dc197d664cf7cc2368500b08ed584a52b4c9398f